File: AndroidTools/AndroidTools/Domain/UseCases/GetPackagesUseCase.swift
//
//  GetPackagesUseCase.swift
//  AndroidTools
//
//  Created by Thomas Bernard on 21/05/2024.
//

import Foundation

class GetPackagesUseCase {
    private let packageRepository : PackageRepository = PackageRepositoryImpl()
    
    func execute(deviceId : String) -> Result<[String], GetAllPackagesError> {
        let result = packageRepository.getAllPackages(deviceId: deviceId)
        return result
    }
}

File: lib/shared/domain/usecases/get_packages_usecase.dart
import 'package:android_tools/shared/domain/repositories/package_repository.dart';

class GetPackagesUsecase {
  final PackageRepository _packageRepository;

  GetPackagesUsecase(this._packageRepository);

  Future<Iterable<String>> call(String deviceId) {
    return _packageRepository.getAllPackages(deviceId);
  }
}
